src/python/TaskWorker/Actions/PreJob.py

```diff
@@ -535,6 +535,17 @@ def execute(self, *args):
         sleep_time = int(self.dag_retry)*60
         if old_time:
             sleep_time = int(max(1, sleep_time - old_time))
+        # Double check if it is HC job as they want to have finer submittion
+        if self.task_ad.get("CRAB_HC", '') and \
+           self.task_ad.get("CRAB_HC_JOB_TIMEOUT", '0') and \
```

Comment:
in any case please wrap this part of code inside a clear
# this is done for Hammer Cloud etc. etc.
because who will read this next year will not know of
the plan you discussed with Stephan which may not be
documented in any permanent place.

as usual... more inline documentation will help, e.g. it takes quite a leap to imagine that 
touch_logs returns a time (from epoch!) Why not make it return nothing and call time in the main if 
you want to know which time it is now ?




On 03/18/2016 02:54 PM, Marco Mascheroni wrote:
> In src/python/TaskWorker/Actions/PreJob.py
> <https://github.com/dmwm/CRABServer/pull/5163#discussion_r56658465>:
>
>> @@ -535,6 +535,17 @@ def execute(self, *args):
>>          sleep_time = int(self.dag_retry)*60
>>          if old_time:
>>              sleep_time = int(max(1, sleep_time - old_time))
>> +        # Double check if it is HC job as they want to have finer submittion
>> +        if self.task_ad.get("CRAB_HC", '') and \
>> +           self.task_ad.get("CRAB_HC_JOB_TIMEOUT", '0') and \
>
> I am not sure I would like to bind this functionality to HC. Let's have a generic name for this
> classad for slow release. Also, when the name is picked up we should document this feature, @emaszs
> <https://github.com/emaszs> can you do it? Maybe here
> https://twiki.cern.ch/twiki/bin/view/CMSPublic/CRAB3TipsAndTricks is a good place @belforte
> <https://github.com/belforte> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dmwm/CRABServer/pull/5163/files/78311507f6f6663b8f2d972b15bf2942271a81c6#r56658465>
>